kernel/src/hil/time.rs

```diff
@@ -35,6 +42,33 @@ pub trait Ticks: Clone + Copy + From<u32> + fmt::Debug + Ord + PartialOrd + Eq {
     /// are 32 bits.
     fn into_u32(self) -> u32;
 
+    /// The amount of bits required to right-pad this timer (filling
+    /// the lower bits with `0`) for it wrap at `(2 ** 32) - 1`
+    /// bits. For timers with a `width` larger than 32, this value
+    /// will be `0` (i.e., they can simply be truncated to 32-bits).
+    fn u32_padding() -> usize {
+        32_usize.saturating_sub(Self::width())
```

Comment:
(pedantic comment) I think this should probably be `u32::BITS` https://doc.rust-lang.org/core/primitive.u32.html#associatedconstant.BITS